### DISCUSSION TITLE
Make image size limit bigger than 256x256

### DISCUSSION THREAD

**Discussion by frutiemax** [OpenRCT2/OpenRCT2#16309] [Feature requests]
It would be very useful to have image with size bigger than 256x256 for bigger flat rides to avoid doing some painting shenanigans.

**Comment by frutiemax** [comment]
Right now, if I disactivate the image limits check, the limitation is the RLE image compression and decompression, as RAW image importing and exporting using the sprite commands using a 1024x1024 work fine. The RLE image compression and decompression needs to encode the number of consecutive pixels into 2 bytes for each pixel, so each pixel should be 3 bytes.

I'd like to know if anyone knows any RLE image compression/decompression that we could use in the future to avoid making a homemade algorithm.

**Comment by duncanspumpkin** [comment]
The limitation of rle images is that they must be less than 0xFFFF bytes. This is due to the jump table at the start of the image data.

**Comment by ZehMatt** [comment]
There are probably way more pitfalls in increasing this limit than visible on the surface, I believe the dimensions are also stored in int8_t. Also isn't large scenery already doing what you want? Exception being multiple sprites are drawn instead of one, which is quite typical for sprite based drawing.

**Reply by ZehMatt** [reply]
Another thing to consider is the atlas that we use in OpenGL, you can't just use any image dimension for this to work.
